commands/run.sh

```diff
@@ -131,8 +136,19 @@ if [[ "$(plugin_read_config DEPENDENCIES "true")" == "false" ]] ; then
   run_params+=(--no-deps)
 fi
 
-if [[ -n "$(plugin_read_config WORKDIR)" ]] ; then
-  run_params+=("--workdir=$(plugin_read_config WORKDIR)")
+workdir=''
+
+if [[ -n "$(plugin_read_config WORKDIR)" ]] || [[ "${mount_checkout}" == "true" ]]; then
+  workdir="$(plugin_read_config WORKDIR "$workdir_default")"
+fi
+
+if [[ -n "${workdir}" ]] ; then
+  run_params+=("--workdir=${workdir}")
+fi
+
+# By default, mount $PWD onto $WORKDIR
+if [[ "${mount_checkout}" == "true" ]] ; then
+  run_params+=("-v" "${pwd_default}:${workdir}")
```

Comment:
basically lifted from https://github.com/buildkite-plugins/docker-buildkite-plugin/blob/391ee61fb87b1281d85e97263fd4a46181265687/hooks/command#L127-L136 and https://github.com/buildkite-plugins/docker-buildkite-plugin/blob/391ee61fb87b1281d85e97263fd4a46181265687/hooks/command#L193-L196 and adapted to this plugin